src/commcare_cloud/ansible/plugins/lookup/cached_aws_secret.py

```diff
@@ -30,3 +37,33 @@ def run(self, terms, variables, **kwargs):
             raise value
         else:
             return value
+
+    def get_cache(self, term, inventory_dir):
+        try:
+            with open(self._secrets_cache_filename(term, inventory_dir=inventory_dir), 'rb') as f:
+                contents = f.read()
+                if contents == b'FETCHING':
+                    time.sleep(1)
+                    return self.get_cache(term, inventory_dir=inventory_dir)
```

Comment:
The logic here is convoluted. It seems like it should be possible to refactor to remove the sleep and recursive call, which should clear it up quite a bit.

Reply:
It's probably possible to remove the recursive call, but I don't think it's possible to remove the sleep (maybe reduce it, but not remove it). What's going on here is that during the execution of a task, ansible makes N forks, where N is the number of machines the task will run on. Each of these forks then more or less simultaneously calls this plugin for secrets, usually in the same order. Instead of N forks collectively making N calls for secret X, we want whichever gets there first to claim that responsibility by writing 'FETCHING' to the cache file, and for all the other forks to wait until a value other than 'FETCHING' appears in the cache file. This results in one request per secret rather than N requests per secret.

Reply:
I took a stab at refactoring this for better readability